localization/localization.go

```diff
@@ -0,0 +1,64 @@
+package localization
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/nicksnyder/go-i18n/v2/i18n"
+	"golang.org/x/text/language"
+)
+
+const (
+	fileExtension = ".json"
+)
+
+var (
+	localizers sync.Map
+)
+
+// RegisterLocale Locales
+func RegisterLocale(locale string, localeDir string) {
+	bundle := i18n.NewBundle(language.MustParse(DefaultLocale))
+	bundle.RegisterUnmarshalFunc("json", json.Unmarshal)
+
+	locale = strings.ToLower(locale)
+	path := filepath.Join(localeDir, locale+fileExtension)
+	bundle.MustLoadMessageFile(path)
+
+	localizers.Store(locale, i18n.NewLocalizer(bundle, locale))
+}
+
+// RegisterLocales Locales
+func RegisterLocales(locales []string, localeDir string) {
+	for _, locale := range locales {
+		RegisterLocale(locale, localeDir)
+	}
+}
+
+func Localize(lang string, msgID string, data interface{}) (string, error) {
```

Comment:
How can golint let you escape with this function without a document 😂 ?

Reply:
Aha, we don't have any githook for this repo 😂

Reply:
updated